src/Catalyst.Core.Modules.Ledger/Ledger.cs

```diff
@@ -256,5 +286,114 @@ public void Dispose()
+            }
+
+            private TransactionReceipt BuildFailedReceipt(Address recipient, long gasSpent)
+            {
+                return BuildReceipt(recipient, gasSpent, StatusCode.Failure, LogEntry.EmptyLogs);
+            }
+
+            private TransactionReceipt BuildReceipt(Address recipient, long spentGas, byte statusCode, LogEntry[] logEntries)
+            {
+                PublicEntry entry = _delta.PublicEntries[_currentIndex];
+
+                TransactionReceipt txReceipt = new TransactionReceipt
+                {
+                    Logs = logEntries,
+                    GasUsedTotal = _delta.GasUsed,
+                    StatusCode = statusCode,
+                    Recipient = entry.IsContractDeployment ? null : recipient,
+                    DeltaHash = _deltaHash,
+                    DeltaNumber = _deltaNumber,
+                    Index = _currentIndex,
+                    GasUsed = spentGas,
+                    Sender = GetAccountAddress(entry.SenderAddress),
+                    ContractAddress = entry.IsContractDeployment ? recipient : null
+                };
+
+                _currentIndex += 1;
+
+                return txReceipt;
+            }
+
+            private static Address GetAccountAddress(ByteString publicKeyByteString)
+            {
+                if (publicKeyByteString == null || publicKeyByteString.IsEmpty)
+                {
+                    return null;
+                }
+
+                return publicKeyByteString.ToByteArray().ToKvmAddress();
+            }
+
+            public bool IsTracingActions => false;
+            public bool IsTracingOpLevelStorage => false;
+            public bool IsTracingMemory => false;
+            public bool IsTracingInstructions => false;
+            public bool IsTracingCode => false;
+            public bool IsTracingStack => false;
+            public bool IsTracingState => false;
+            public void ReportBalanceChange(Address address, UInt256? before, UInt256? after) { throw new NotImplementedException(); }
+            public void ReportCodeChange(Address address, byte[] before, byte[] after) { throw new NotImplementedException(); }
+            public void ReportNonceChange(Address address, UInt256? before, UInt256? after) { throw new NotImplementedException(); }
+            public void ReportStorageChange(StorageAddress storageAddress, byte[] before, byte[] after) { throw new NotImplementedException(); }
+            public void StartOperation(int depth, long gas, Instruction opcode, int pc) { throw new NotImplementedException(); }
+            public void ReportOperationError(EvmExceptionType error) { throw new NotImplementedException(); }
+            public void ReportOperationRemainingGas(long gas) { throw new NotImplementedException(); }
+            public void SetOperationStack(List<string> stackTrace) { throw new NotImplementedException(); }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'SetOperationStack'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)